add docs for google secret rotation (#1996)
nginx/documentation: content/nginxaas-google/getting-started/ssl-tls-certificates/overview.md

## Add SSL/TLS certificates

NGINXaaS supports two ways to manage your certificates and keys securely:

**NGINXaaS console** — Manage certificates alongside the NGINX configurations that reference them. See [Add certificates using the NGINXaaS Console]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-console.md" >}}).

**Google Secret Manager** — Fetch secrets directly from [Secret Manager](https://docs.cloud.google.com/secret-manager/docs/overview), keeping credentials within Google Cloud. See [Add certificates from Secret Manager]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-secret-manager.md" >}}).

## Certificate rotation

NGINXaaS supports automatic and manual rotation for Secret Manager certificates:

**Automatic rotation** — Let NGINXaaS pick up new certificate versions automatically with no configuration changes needed. See [Rotate a Secret Manager certificate (automatic)]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-secret-manager.md#rotate-a-secret-manager-certificate-automatic" >}}).

> **Manual rotation** — When you need an immediate update, use **Reapply Configuration** in the console to refetch secrets right away. See [Rotate a Secret Manager certificate (manual)]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-secret-manager.md#rotate-a-secret-manager-certificate-manual" >}}).

Reviewer: nit (clarification on what's being updated):

```suggestion
**Manual rotation** — When you need to update certificates immediately, use **Reapply Configuration** in the console to refetch secrets right away. See [Rotate a Secret Manager certificate (manual)]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-secret-manager.md#rotate-a-secret-manager-certificate-manual" >}}).
```

- [Add certificates from Secret Manager]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-secret-manager.md" >}})
- [Add certificates using the NGINXaaS Console]({{< ref "/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-console.md" >}})

> ## Rotate a Secret Manager certificate (manual)

Reviewer 2: All of this new content is specific to Google Secret Manager managed certificates. It seems it would fit better as topics within `/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-secret-manager.md` rather than this overview page.

Reviewer: I've been seeing the overview pages as "info" pages that describes what the feature is. Then, the other pages are more "how to" pages that describe how to enable the feature. (@JTorreG / @mjang feel free to tell me I'm off base)

With that in mind, I think it makes sense to describe manual and autorotation in the overview page and link to the "Add certificates from Secret Manager" how-to page. That way, the feature is more visible (maybe that's subjective)

Reviewer 2: Yeah, this content sounded more like "how to rotate Secret Manager certs" not "what is autorotation".

I could see us expanding this overview a bit more to provide some info about each kind of cert management "tool". eg.

```
There are multiple ways to manage your SSL/TLS certs and keys securely:

*Google Secret Manager* can be utilized to fetch secrets directly from [Secret Manager](https://docs.cloud.google.com/secret-manager/docs/overview), without ever leaving Google Cloud infrastructure. This solution allows you to use secret aliases for auto-rotation, or secret version pinning for greater control of cert rotation through your deployments.

*NGINXaaS Console* can be used to manage your certs and keys right next to your NGINXaaS configurations that use them. [...]
```

Reviewer: Yeah I like that 👍 

@vivki thoughts on moving the more informational bits to the overview page and leave the how-to instructions in `/nginxaas-google/getting-started/ssl-tls-certificates/ssl-tls-certificates-secret-manager.md`?

> To immediately refetch secrets without editing your NGINX configuration, use **Sync Configuration**. This is useful in two scenarios:

Reviewer 3: In case we get more than two scenarios

```suggestion
To immediately refetch secrets without editing your NGINX configuration, use **Sync Configuration**. This is useful in the following scenarios:
```

- **New secret version**: You've uploaded a new certificate and want NGINXaaS to use it right away.

> - **WIF or permissions fix**: You've updated a WIF provider or granted Secret Manager permissions and want NGINXaaS to retry immediately.

Reviewer 3: What is WIF? This is the first time we're using the acronym on this page

Author: edited to add "Workload Identity Federation (WIF)"

To reapply your configuration:

1. In the NGINXaaS console, go to your deployment.
2. Open the **Configuration Info** panel.
3. Select **Reapply Configuration**.

NGINXaaS reapplies your current configuration version and immediately refetches all referenced secrets.

## Rotate a Secret Manager certificate (automatic)

If your configuration references secrets using the `versions/latest` alias, NGINXaaS automatically picks up new certificate versions. When you add a new secret version in Secret Manager, the next scheduled sync fetches it and reloads NGINX. No configuration changes are required.

### Rotation timeline

> New secret versions are usually applied to your deployment within approximately 4 hours.

Reviewer 4: we should avoid vague language like "usually" and "approximately"

can this be reworded to say something like "on average..."

Reviewer 2: Or be more assertive, since this is the requirement, right?

```suggestion
New secret versions are picked up and applied to your deployment within four hours.
```

To rotate your certificate:

1. Generate your new certificate and key.
2. Add the updated certificate and key as new secret versions in Secret Manager. No configuration changes are needed.
3. Wait for NGINXaaS to apply the updated certificate. Rotation is complete when the new certificate serial number appears on your NGINX endpoint and in the NGINXaaS **Certificates** list.

{{< call-out "note" >}}

> 
> Automatic rotation only works when the Google Secret ID uses the `latest` version alias. If you've pinned a specific version number, manually rotate using **Reapply Configuration**.
> 

Reviewer 4: ```suggestion
Automatic rotation only works when the Google Secret ID uses the `latest` version alias. If you've pinned a specific version number, manually rotate using **Reapply Configuration**.
```

{{< /call-out >}}

## Monitor secret fetch events

> NGINXaaS generates an event each time it fetches — or fails to fetch — a certificate or key from Secret Manager. Use these events to track successful rotations and troubleshoot access failures.

Reviewer 4: ```suggestion
NGINXaaS generates an event each time it fetches or fails to fetch a certificate or key from Secret Manager. Use these events to track successful rotations and diagnose access failures.
```

### Event types

{{< table >}}

| Event type | Description |
|---|---|
| Successful Secret Fetch from Google | The secret was fetched from Secret Manager and applied to NGINX. |
| Failed Secret Fetch from Google | NGINXaaS couldn't fetch the secret. The event message includes the error details. |

> 

Reviewer 4: ```suggestion
| Event type | Description |
|---|---|
| Successful Secret Fetch from Google | The secret was fetched from Secret Manager and applied to NGINX. |
| Failed Secret Fetch from Google | NGINXaaS couldn't fetch the secret. The event message includes the error details. |
```

don't leave blank lines between the shortcode tags and the content

{{< /table >}}

### View events in the console

- Select **Overview** in the left menu, then select **Events**.
- To narrow results to a specific deployment, filter by its object ID using the controls at the top of the page.

> - For a summary of recent events, select **Deployments**, select the deployment, and look for the **Recent Events** card. Select **See Events Details** to go to the full Events page pre-filtered for that deployment.

Reviewer 3: Should these be numbered? They seem sequential

Author: The 1st + 2nd steps are Events page workflow, the 3rd is a different workflow--navigating to the Event Info from the specific deployment page. Not necessarily sequential so leaving the step format.

### Common failure messages and remediation

{{< table >}}

| Message | Likely cause | Remediation |
|---|---|---|
| `Failed to fetch secret ... PermissionDenied: Permission 'secretmanager.versions.access' denied` | The Workload Identity Federation principal doesn't have the required IAM role on the secret. | Verify the WIF principal has the Secret Manager Secret Accessor role on the project or secret. |
| `Failed to fetch secret ... NotFound: Secret [...] has no alias [latest]` | No versions exist for the referenced secret, or the specified version alias or number doesn't exist. | Confirm the secret has at least one enabled version and that the resource name in your configuration uses a valid version or alias. |

> {{< /table >}}

Reviewer 4: ```suggestion
{{< table >}}
| Message | Likely cause | Remediation |
|---|---|---|
| `Failed to fetch secret ... PermissionDenied: Permission 'secretmanager.versions.access' denied` | The Workload Identity Federation principal doesn't have the required IAM role on the secret. | Verify the WIF principal has the Secret Manager Secret Accessor role on the project or secret. |
| `Failed to fetch secret ... NotFound: Secret [...] has no alias [latest]` | No versions exist for the referenced secret, or the specified version alias or number doesn't exist. | Confirm the secret has at least one enabled version and that the resource name in your configuration uses a valid version or alias. |
{{< /table >}}
```
